src/routines/write_model.jl

```diff
@@ -1,70 +1,75 @@
+struct OptimizerLog
+
+    termination_status::Enum
+    primal_status::Enum
+    dual_status::Enum
+    solve_time::Float64
+
+end
+
+
 # taking the outputted files for the variable DataFrame and writing them to a featherfile
-function _write_variable_results(vars_results::Dict{Symbol, DataFrames.DataFrame}, save_path::AbstractString)
+function write_data(vars_results::Dict{Symbol, DataFrames.DataFrame}, save_path::AbstractString; kwargs...)
 
+    file_type = get(kwargs, :file_type, Feather)
     for (k,v) in vars_results
-         file_path = joinpath(save_path,"$(k).feather")
-         Feather.write(file_path, vars_results[k])
+         file_path = joinpath(save_path,"$(k).$file_type")
+         file_type.write(file_path, vars_results[k])
     end
 
     return
-
 end
 
-function _write_variable_results(canonical::CanonicalModel, save_path::AbstractString)
 
-    for (k,v) in vars(canonical)
-         file_path = joinpath(save_path,"$(k).feather")
-         Feather.write(file_path, _result_dataframe_vars(v))
+function write_data(vars_results::OperationModel, save_path::AbstractString; kwargs...)
+
+    file_type = get(kwargs, :file_type, Feather)
+    for (k,v) in vars(vars_results.canonical)
+         file_path = joinpath(save_path,"$(k).$file_type")
+         file_type.write(file_path, _result_dataframe_vars(v))
+
     end
 
     return
-
 end
 
-function _write_optimizer_log(optimizer_log::Dict{Symbol, Any}, save_path::AbstractString)
-
-    optimizer_log[:termination_status] = Int(optimizer_log[:termination_status])
-    optimizer_log[:primal_status] = Int(optimizer_log[:primal_status])
-    optimizer_log[:dual_status] = Int(optimizer_log[:dual_status])
-    optimizer_log[:solve_time] = optimizer_log[:solve_time]
+function write_data(data::DataFrames.DataFrame, save_path::AbstractString, file_name::String; kwargs...)
```

Comment:
Some users might not notice that file_name should not have an extension.  You could add a docstring or strip it if they add one.